test/signal_test.dart

```diff
@@ -0,0 +1,385 @@
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(await output, 'SIGTERM\nbye!');
+      expect(await script.kill(), false);
+    });
+  });
+
+  group('on Script.capture', () {
+    test("does not interrupt by default", () async {
+      var script = Script.capture((_) async {
+        await Future<void>.delayed(Duration(seconds: 3));
+        print('done!');
+      });
+      var output = script.output;
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(await output, 'done!');
+      expect(await script.kill(), false);
+    });
+
+    test("can be captured with onSignal", () async {
+      late Script innerScript;
+      var script = Script.capture((_) async {
+        innerScript = _watchSignalsAndExit();
+        await innerScript.done;
+      }, onSignal: (signal) => innerScript.kill(signal));
+      var output = script.output;
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(await output, 'SIGTERM\nbye!');
+      expect(await script.kill(), false);
+    });
+  });
+
+  group('BufferedScript.capture', () {
+    test("does not interrupt by default", () async {
+      var script = BufferedScript.capture((_) async {
+        await Future<void>.delayed(Duration(seconds: 3));
+        print('done!');
+      });
+      var done = false;
+
+      var output = script.output.then((v) {
+        done = true;
+        return v;
+      });
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(done, false);
+      await script.release();
+
+      expect(await output, 'done!');
+      expect(await script.kill(), false);
+    });
+
+    test("can be captured with onSignal", () async {
+      late Script innerScript;
+      var script = BufferedScript.capture((_) async {
+        innerScript = _watchSignalsAndExit();
+        await innerScript.done;
+      }, onSignal: (signal) => innerScript.kill(signal));
+
+      var done = false;
+      var output = script.output.then((v) {
+        done = true;
+        return v;
+      });
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(await script.kill(), true);
+
+      expect(done, false);
+      await script.release();
+
+      expect(await output, 'SIGTERM\nbye!');
+      expect(await script.kill(), false);
+    });
+  });
+
+  test("interrupts a Script.fromLineTransformer", () async {
+    var script = Script.fromLineTransformer(StreamTransformer.fromHandlers(
+      handleData: (data, sink) => sink.add(data.toUpperCase()),
+    ));
+    Object? error;
+    script.done.catchError((Object e) => error = e);
+    var events = <String>[];
+    var done = false;
+    script.stdout.lines.listen(events.add, onDone: () => done = true);
+
+    script.stdin.writeln('before');
+    await Future<void>.delayed(Duration(seconds: 1));
```

Comment:
For uses like this where you're just trying to let the event queue run dry, you can use [`pumpEventQueue()`](https://pub.dev/documentation/test_api/latest/hooks/pumpEventQueue.html).